### PR TITLE
SALTO-1533: Avoid sharing filter context between deploy calls

### PR DESCRIPTION
**Pull request by ori-moisis**
This changes the usage pattern of the filters runner such that it will not allow
filters to accidentally share context between preDeploy and onDeploy of different
deploy calls

---

The problematic scenario was where two (or more) concurrent `adapter.deploy` calls were made, this could lead to the same filter instance getting called with `preDeploy` for deploy A, then `preDeploy` for deploy B before returning to `onDeploy` of deploy A.
for stateful filters this caused an issue since they rely on their state in `onDeploy` to match what they saw in `preDeploy`.

Changing the usage pattern such that every `adapter.deploy` call creates a new set of filter instances resolves this issue

---
_Release Notes_: 
Salesforce adapter:
- Fixed an issue where the workspace state was not properly updated after a `deploy` call that involved both metadata and data records

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/salesforce-adapter/src/client/jsforce.ts [hunk 1]
```diff
@@ -43,7 +42,7 @@ export interface Metadata {
   retrieve(request: RetrieveRequest,
     callback?: Callback<RetrieveResult>): RetrieveResultLocator<RetrieveResult>
   deploy(
-    zipInput: Stream | Buffer | string, options: DeployOptions
+    zipInput: Buffer | string | NodeJS.ReadableStream, options: DeployOptions
```
Comment (netama) [inline packages/salesforce-adapter/src/client/jsforce.ts:45]: seems unrelated?

**Review by netama — APPROVED** [review]
LGTM

File: packages/salesforce-adapter/src/client/jsforce.ts (hunk 1 above)
Comment (ori-moisis) [reply packages/salesforce-adapter/src/client/jsforce.ts:45]: needed it in one of the tests, then I checked the jsforce code and saw that this is a more accurate type description
